### PR TITLE
Feat/signup

### PR DESCRIPTION
**Pull request by tmznwnel00**
SIgnup with POST method (password hash)
Login with Post method (return JWT token)

reference link : https://fastapi.tiangolo.com/tutorial/security/oauth2-jwt/
Above link contains hashing password and handling JWT

### CODE REVIEW

File: server/database.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from sqlalchemy import create_engine
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
+user = 'test'
+password = 'Abc1234!'
```
Comment (sungminoh) [inline server/database.py]: Remove this part and use env variables for better security.

File: server/main.py [hunk 2]
```diff
@@ -1,7 +1,101 @@
...
+from typing import Union
+
+from jose import JWTError, jwt
+from passlib.context import CryptContext
+from datetime import datetime, timedelta
+
+model.Base.metadata.create_all(bind=engine)
+
+# Acquired by CLI "openssl rand -hex 32"
+SECRET_KEY = "9ff8f5a166924e4cc01ad9682302abf54ac778cf794cbf642074bf3ed58ccaf0"
```
Comment (sungminoh) [inline server/main.py:17]: This should be secret
ref to https://github.com/tymondesigns/jwt-auth/issues/311

File: server/main.py [hunk 3]
```diff
@@ -1,7 +1,101 @@
...
+
+def get_user(db, username):
+    return db.query(model.User).filter(model.User.login_id == username).first()
+    
+def get_db():
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
```
Comment (sungminoh) [inline server/main.py]: Move function part to a DB related module(e.g. `database.py`) since this will be reused by many other APIs.

**Review by sungminoh — APPROVED** [review]
If you create a model like `secret.py` or `constants.py` and put all variables that need to be changed later, it will be easier to manage.

**Comment by tmznwnel00** [comment]
> If you create a model like `secret.py` or `constants.py` and put all variables that need to be changed later, it will be easier to manage.

I didn't understand this comment.
Is there specific part I should put variables to that models?

File: server/main.py (hunk 2 above)
Comment (tmznwnel00) [reply server/main.py:17]: @sungminoh SECRET_KEY to .env file?

Comment (sungminoh) [reply server/main.py:17]: yes.

**Comment by sungminoh** [comment]
> > If you create a model like `secret.py` or `constants.py` and put all variables that need to be changed later, it will be easier to manage.
> 
> I didn't understand this comment. Is there specific part I should put variables to that models?

My bad. There was a typo. You can create a file named `secret.py` or `constants.py` wherever you want (root or under utils directory).
You can define `DB_USERNAME = XXX` and `DB_PASSWORD = YYY` in that file and import the module wherever you want to use the variable.
In this way, you can change them to `env.get('<ENV_NAME>'` easily without going through hassle of finding all usage.

**Pull request merged**